src/libtexture/imagecache_pvt.h

```diff
@@ -709,21 +708,16 @@ class ImageCachePerThreadInfo {
     }
 
     // Add a new filename/fileptr pair to our microcache
-    void filename(ustring n, ImageCacheFile* f)
+    void remember_filename(ustring n, ImageCacheFile* f)
     {
-        last_filename[next_last_file] = n;
-        last_file[next_last_file]     = f;
-        ++next_last_file;
-        next_last_file %= nlastfile;
+        m_thread_files[n] = f;
```

Comment:
`m_thread_files.emplace(n, f)` may be a smidge more efficient

Reply:
Sure, I'll change to that.

Reply:
pushed an update with this fix